turbo/stages/headerdownload/header_algos.go

```diff
@@ -667,7 +667,7 @@ func (hd *HeaderDownload) RequestMoreHeaders(currentTime uint64) (*HeaderRequest
 			// Produce a header request that would extend this anchor (add parent, parent of parent, etc.)
 			return &HeaderRequest{
 				Anchor:  anchor,
-				Hash:    anchor.parentHash,
+				Hash:    common.Hash{},
```

Comment:
This doesn't look right to me. Hash is more specific than the block number, we should use hash. Moreover, we don't know the block number when we receive a forkChoiceUpdated with a missing parent hash.